### DISCUSSION TITLE
brightnessctl support added to the backlight module

### DISCUSSION THREAD

**Discussion by QingyaoLin** [polybar/polybar#2914] [Ideas]
Currently, the backlight module seems to require light in order to function properly. brightnessctl is much more convenient than light, it doesn't require any extra setup, it works right out of the box, and it can work on X or on Wayland.

**Comment by patrick96** [comment]
The brightness module doesn't need any external software to function, it directly reads backlight information out of `/sys/class/backlight`.

It only requires extra setup if you want to change the brightness through the module (both light and brightnessctl also require the same setup, they can just do it automatically during the installation).

Also, the contents of `/sys/class/backlight` should be independent of your display server, so it should work on both X and Wayland (of course polybar has no wayland support though)

**Comment by QingyaoLin** [comment]
However, the backlight module only supports light out of the box, and once I switched it to brightnessctl, the tuning failed.

This means that once I am living in both X and Wayland, I will have to install both light (It doesn't seem to support wayland) and brightnessctl.  I'm sorry, it's a quirk.

**Reply by patrick96** [reply]
So when you don't have light installed, the [backlight module](https://github.com/polybar/polybar/wiki/Module:-backlight) does not work at all? Does it print an error message?

**Reply by QingyaoLin** [reply]
It can't adjust the brightness with the middle mouse button.

**Reply by QingyaoLin** [reply]
polybar|info:  Delivered action to 1 module
polybar|info:  Forwarding action to modules (module: 'backlight', action: 'dec', data: '')
polybar|info:  module/backlight: Changing value by -5%
polybar|error: module/backlight: Unable to change backlight value. Your system may require additional configuration. Please read the module documentation.
(reason: failed to write to /sys/class/backlight/intel_backlight/brightness: Permission denied)
polybar|info:  Delivered action to 1 module
polybar|info:  Forwarding action to modules (module: 'backlight', action: 'dec', data: '')
polybar|info:  module/backlight: Changing value by -5%
polybar|error: module/backlight: Unable to change backlight value. Your system may require additional configuration. Please read the module documentation.
(reason: failed to write to /sys/class/backlight/intel_backlight/brightness: Permission denied)

**Reply by QingyaoLin** [reply]
However, in the terminal, I can `brightlnessctl set 5%-`.

**Reply by QingyaoLin** [reply]
> However, in the terminal, I can `brightlnessctl set 5%-`.

This does not require permission.

**Reply by patrick96** [reply]
Yes, the backlight module requires extra configuration if you want it to be able to change the brightness, because the user needs to be able to change the content of the brightness file.

The [arch wiki](https://wiki.archlinux.org/title/Backlight#ACPI) has some good information for how to do that.
